src/main/java/com/ltonetwork/client/core/PublicNode.java

```diff
@@ -0,0 +1,102 @@
+package com.ltonetwork.client.core;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ltonetwork.client.core.transacton.*;
+import com.ltonetwork.client.exceptions.BadMethodCallException;
+import com.ltonetwork.client.utils.HttpClientUtil;
+import com.ltonetwork.client.utils.JsonObject;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpResponse;
+import java.util.Map;
+
+public class PublicNode {
+    private final String url;
+    private final String apiKey;
+
+    public PublicNode(String url, String apiKey) {
+        this.url = url;
+        this.apiKey = apiKey;
+    }
+
+    public String getUrl() {
+        return url;
+    }
+
+    public String getApiKey() {
+        return apiKey;
+    }
+
+    public Transaction getTransaction(int id) throws URISyntaxException {
+        HttpResponse<String> resp = HttpClientUtil.get(new URI(String.format("%s/transactions/info/%d", this.url, id)));
+        return getTransactionObject(new JsonObject(resp.body()));
+    }
+
+    public Transaction getUnconfirmed() throws URISyntaxException {
+        HttpResponse<String> resp = HttpClientUtil.get(new URI(String.format("%s/transactions/unconfirmed", this.url)));
+        return getTransactionObject(new JsonObject(resp.body()));
+    }
+
+    public JsonObject compile(String script) throws URISyntaxException {
```

Comment:
Let `compile` add a SetScript transaction. Maybe add the estimated costs to the object. The method for getting the estimated costs should throw an exception if the object wasn't created through `compile`.